src/test/java/io/druid/guice/ConditionalMultibindTest.java

```diff
@@ -0,0 +1,257 @@
+package io.druid.guice;
+
+import com.google.common.base.Predicates;
+import com.google.common.collect.ImmutableSet;
+import com.google.inject.Binder;
+import com.google.inject.BindingAnnotation;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import com.google.inject.Key;
+import com.google.inject.Module;
+import com.google.inject.TypeLiteral;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import java.util.Properties;
+import java.util.Set;
+
+/**
+ */
+public class ConditionalMultibindTest
+{
+
+  private static final String ANIMAL_TYPE = "animal.type";
+
+  private Properties props;
+
+  @Before
+  public void setUp() throws Exception
+  {
+    props = new Properties();
+  }
+
+  @Test
+  public void testMultiConditionalBind_cat()
+  {
+    props.setProperty("animal.type", "cat");
+
+    Injector injector = Guice.createInjector(new Module()
+    {
+      @Override
+      public void configure(Binder binder)
+      {
+        ConditionalMultibind.create(props, binder, Animal.class)
+                            .conditionBinder(ANIMAL_TYPE, Predicates.equalTo("cat"), Cat.class)
+                            .conditionBinder(ANIMAL_TYPE, Predicates.equalTo("dog"), Dog.class);
+      }
+    });
+
+    Set<Animal> animalSet = injector.getInstance(Key.get(new TypeLiteral<Set<Animal>>(){}));
```

Comment:
if you have an object like 


```java
class PetShopAvails {
    @Inject
    Set<Animals> animals;
}
```

and call 

```java
PetShopAvails shop = new PetShopAvails();
injector.inject(shop);
```

Does it fill in the animals set appropriately?